packages/components/src/SegmentedControl/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useId, useMemo } from "react";
+import { useSegmentedControl } from "./SegmentedControlProvider";
+
+interface SegmentedControlOptionProps extends PropsWithChildren {
+  /**
+   * The value of the option.
+   */
+  readonly value: string | number;
+  /**
+   * The label to appear beside the radio button.
+   */
+  readonly checked?: boolean;
+  readonly label?: string;
+  readonly onChange?: (valueIn: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
Nit: argument name is actually the native event so it's a bit misleading. Do we want to expose the native event or should we just have a `onSelected` callback that fires.